code/game/objects/items/motion_detector.dm

```diff
@@ -1,353 +1,149 @@
-
-#define MOTION_DETECTOR_LONG 0
-#define MOTION_DETECTOR_SHORT 1
-
-#define MOTION_DETECTOR_HOSTILE ""
-#define MOTION_DETECTOR_FRIENDLY "_friendly"
-#define MOTION_DETECTOR_DEAD "_dead"
-#define MOTION_DETECTOR_FUBAR "_fubar" //i.e. can't be revived. Might have useful gear to loot though!
-
-
-/obj/effect/detector_blip
+/obj/effect/edge_blip
 	icon = 'icons/Marine/marine-items.dmi'
-	icon_state = "detector_blip"
-	layer = BELOW_FULLSCREEN_LAYER
+	plane = ABOVE_HUD_PLANE
+	/// A friendly/hostile identifier
 	var/identifier = MOTION_DETECTOR_HOSTILE
-	var/edge_blip = FALSE
-	var/turf/center
-	var/turf/target_turf
-
-/obj/effect/detector_blip/friendly
-	icon_state = "detector_blip_friendly"
-	identifier = MOTION_DETECTOR_FRIENDLY
 
-/obj/effect/detector_blip/dead
-	icon_state = "detector_blip_dead"
-	identifier = MOTION_DETECTOR_DEAD
+/obj/effect/edge_blip/Initialize(mapload, identifier, mob/operator, screen_pos_x, screen_pos_y, direction = SOUTH)
+	. = ..()
+	if(!operator?.client)
+		qdel(src)
+		return
```

Comment:
this should be return hint initialize qdel or something like that if i remember.